awx/main/tasks/system.py

```diff
@@ -518,6 +518,12 @@ def inspect_execution_and_hop_nodes(instance_list):
 
         inspect_established_receptor_connections(mesh_status)
 
+        for instance in instance_list:
+            if instance.node_type == Instance.Types.HOP and mesh_status['KnownConnectionCosts'].get(instance.hostname):
+                if instance.node_state in (Instance.States.UNAVAILABLE, Instance.States.INSTALLED):
+                    logger.warning(f'Hop node {instance.hostname}, has joined the receptor mesh')
+                instance.save_health_data(errors='', update_last_seen=True)
```

Comment:
Later on in this method we do

```python
instance.save(update_fields=['last_seen'])
```

This is the biggest problem I have right now about how this is evolving. The precedent leans towards the notion that _health checks_ and _updating last seen_ are separate processes. The `update_last_seen` kwarg may make sense for `instance.mark_offline` but it doesn't seem to make sense as something tacked-on to `instance.save_health_data`. The implication following from that is the `save_health_data` saves the data from a health check. If a health check (in the formally proper sense of running `ansible-runner worker --worker-info`) is not run, I'd rather not call it at all.

I also _definitely_ don't want it get set to `now()` for hop nodes if it is set to `parse_date(ad['Time'])` for execution nodes. Does the "KnownConnectionCosts" not report a timestamp? Do nodes not report a timestamp if they're not running a control service? That may be a larger problem with the philosophy.